### PR TITLE
Allow quality reports to regular users

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

- Allowed quality report computation for tasks to regular users (before this only admins could trigger custom report computations)
- Refactored quality control tests

### How has this been tested?

Unit tests

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Release Notes

- **New Features**
  - Regular users can now generate quality reports, expanding access to this functionality.
  - New API endpoints for managing cloud storages, webhooks, and quality reports have been introduced.

- **Improvements**
  - Enhanced permission handling for quality reports, including task-specific checks.
  - Improved job scheduling and reporting mechanisms for quality check jobs.

- **Bug Fixes**
  - Adjusted logic for user permissions related to tasks and projects, ensuring accurate access control.

- **Tests**
  - Added new tests to verify the creation of quality reports by users in different task contexts.

### CODE REVIEW

File: cvat/apps/quality_control/quality_reports.py [hunk 1]
```diff
@@ -2216,8 +2241,8 @@
 
         return rq_job
 
-    def is_custom_quality_check_job(self, rq_job) -> bool:
-        return rq_job.meta.get("job_type") == self._RQ_CUSTOM_QUALITY_CHECK_JOB_TYPE
+    def is_custom_quality_check_job(self, rq_job: RqJob) -> bool:
+        return isinstance(rq_job.id, str) and rq_job.id.startswith(self._QUEUE_CUSTOM_JOB_PREFIX)
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply cvat/apps/quality_control/quality_reports.py:2244]: `isinstance(rq_job.id, str)` already checks for `None` values.

[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply cvat/apps/quality_control/quality_reports.py:2244]: Please close other similar suggestions in this PR.

File: cvat/apps/quality_control/rules/conflicts.rego [hunk 2]
```diff
@@ -41,6 +42,42 @@ import data.organizations
...
+}
+
+is_project_staff if {
+    is_project_owner
+}
+
+is_project_staff if {
+    is_project_assignee
+}
+
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply cvat/apps/quality_control/rules/conflicts.rego:68]: Please close other similar suggestions in this PR.

File: cvat/apps/quality_control/rules/quality_reports.rego [hunk 3]
```diff
@@ -41,6 +42,42 @@ import data.organizations
...
+}
+
+is_project_owner if {
+    input.resource.project.owner.id == input.auth.user.id
+}
+
+is_project_assignee if {
+    input.resource.project.assignee.id == input.auth.user.id
+}
+
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply cvat/apps/quality_control/rules/quality_reports.rego]: Refactored

File: cvat/apps/quality_control/rules/quality_utils.rego [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package quality_utils
+
+import rego.v1
+
+
+is_task_owner(task_data, auth_data) if {
+    task_data.owner.id == auth_data.user.id
+}
```
Comment (zhiltsov-max) [inline cvat/apps/quality_control/rules/quality_utils.rego:8]: Need to move to engine at some point

File: cvat/apps/quality_control/rules/conflicts.rego [hunk 5]
```diff
@@ -58,28 +57,6 @@ allow if {
...
-}
-
-allow if {
-    input.scope == utils.VIEW
-    quality_utils.is_task_staff(input.resource.task, input.resource.project, input.auth)
-    input.auth.organization.id == input.resource.organization.id
-    utils.has_perm(utils.WORKER)
-    organizations.has_perm(organizations.WORKER)
-}
-
```
Comment (Marishka17) [inline cvat/apps/quality_control/rules/conflicts.rego]: @zhiltsov-max, I've not found any place where these rules are used. Could you let me know why you added them? Are you going to introduce the GET `/api/quality/conflicts/<id>` endpoint?

**Pull request merged**